feature_engine/timeseries/forecasting/window_features.py

```diff
@@ -0,0 +1,333 @@
+
+        LagFeatures works only with numerical variables. You can pass a list of variables
+        to lag. Alternatively, LagFeatures will automatically select and lag all numerical
+        variables found in the training set.
+
+        More details in the :ref:`User Guide <lag_features>`.
+
+        Parameters
+        ----------
+        {variables}
+
+        periods: int, list of ints, default=1
+            Number of periods to shift. Can be a positive integer or list of positive
+            integers. If list, features will be created for each one of the periods in the
+            list. If the parameter `freq` is specified, `periods` will be ignored.
+
+        freq: str, list of str, default=None
+            Offset to use from the tseries module or time rule. See parameter `freq` in
+            pandas `shift()`. It is the same functionality. If freq is a list, lag features
+            will be created for each one of the frequency values in the list. If freq is not
+            None, then this parameter overrides the parameter `periods`.
+
+        sort_index: bool, default=True
+            Whether to order the index of the dataframe before creating the lag features.
+
+        {missing_values}
+
+        {drop_original}
+
+        Attributes
+        ----------
+        variables_:
+            The group of variables that will be lagged.
+
+        {feature_names_in_}
+
+        {n_features_in_}
+
+        Methods
+        -------
+        {fit}
+
+        transform:
+            Add lag features.
+
+        {fit_transform}
+
+        See Also
+        --------
+        pandas.shift
+        """
+
+    def __init__(
+            self,
+            variables: List[str] = None,
+            window: Union[str, int] = 1,
+            function: Callable = np.mean,
+            periods: int = 1,
+            freq: str = None,
+            sort_index: bool = True,
+            missing_values: str = "raise",
+            drop_original: bool = False,
+    ) -> None:
+
+        if not (isinstance(window, int) or isinstance(window, str)) or isinstance(window, bool):
+            raise ValueError(
+                f"window must be a string or integer. Got {window} instead."
+            )
+
+        if isinstance(periods, int) and periods > 0 and not isinstance(periods, bool):
+            self.periods = periods
+        else:
+            raise ValueError(
+                f"periods must be a positive integer. Got {periods} instead."
+            )
+
+        if function not in (np.mean, np.std, np.median, np.min, np.max, np.sum):
+            raise ValueError(
+                f"function must be np.mean, np.std, np.median, np.min, np.max, or np.sum. "
+                f"Got {function} instead."
+            )
+        if periods is None and not isinstance(freq, str):
+            raise ValueError(
+                f"freq must be a string. Got {freq} instead."
+            )
+
+        if missing_values not in ["raise", "ignore"]:
```

Comment:
this parameter and the next one are also in lag features. I would make them part of a base class, see my next comment.

Reply:
Just to confirm, are  `missing_values` and `drop_original` the only init params to be included in BaseForecast init?

Reply:
i noticed that variables should also be part of the common tests. Added that in https://github.com/Morgan-Sell/feature_engine/pull/9